src/AutoMapper/MapperConfiguration.cs

```diff
@@ -428,7 +480,20 @@ public LockingConcurrentDictionary(Func<TKey, TValue> valueFactory)
 
         public TValue GetOrAdd(TKey key)
         {
-            return _dictionary.GetOrAdd(key, _valueFactory).Value;
+            try
+            {
+                return _dictionary.GetOrAdd(key, _valueFactory).Value;
+            }
+            catch (InvalidOperationException e) when(e.Message == "ValueFactory attempted to access the Value property of this instance.")

```

Comment:
Yea the circular reference causes and exception with trying to get a Value from Lazy object in LockingDictionary.  Cause you are trying to get a value, while you are already trying to actively resolve said value.  The reason I don't ignore just this exception is because there's a test that expects to throw InvalidOperationException, so I filtered by exception message.